### PR TITLE
feat: add partial support for query temporal entities

### PR DESCRIPTION
**Pull request by HoucemKacem**

### CODE REVIEW

**Comment by HoucemKacem** [comment]
@bobeal I added a commit and it's ready for review.

File: shared/src/main/kotlin/com/egm/stellio/shared/util/ApiUtils.kt [hunk 1]
```diff
@@ -81,10 +82,14 @@ fun hasValueInOptionsParam(options: Optional<String>, optionValue: OptionsParamV
         .filter { it.any { option -> option == optionValue.value } }
         .isPresent
 
-fun parseAndExpandAttrsParameter(attrsParam: String?, contextLink: String): Set<String> =
-    attrsParam
+fun parseRequestParameterAsSet(requestParam: String?): Set<String> =
```
Comment (bobeal) [inline shared/src/main/kotlin/com/egm/stellio/shared/util/ApiUtils.kt:85]: the `AsSet` information in the function name is in the signature of the function

Comment (bobeal) [reply shared/src/main/kotlin/com/egm/stellio/shared/util/ApiUtils.kt:85]: and it won't work with any request parameter ... how to make explicit what you're expecting?

File: search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityService.kt [hunk 2]
```diff
@@ -7,10 +7,18 @@ import org.springframework.stereotype.Service
...
 
 @Service
 class TemporalEntityService {
 
+    fun buildTemporalEntities(
+        queryResult: QueryTemporalEntitiesResult,
+        temporalQuery: TemporalQuery,
+        contexts: List<String>,
+        withTemporalValues: Boolean
+    ) = queryResult.map { buildTemporalEntity(it.first, it.second, temporalQuery, contexts, withTemporalValues) }
```
Comment (bobeal) [inline search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityService.kt:20]: add explicitely the return type of the function (harder to read either)

File: search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityService.kt [hunk 3]
```diff
@@ -7,10 +7,18 @@ import org.springframework.stereotype.Service
 import java.net.URI
 
 typealias SimplifiedTemporalAttribute = Map<String, Any>
+typealias QueryTemporalEntitiesResult = List<Pair<URI, Map<TemporalEntityAttribute, List<AttributeInstanceResult>>>>
```
Comment (bobeal) [inline search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityService.kt:10]: a interesting typealias would be `Map<TemporalEntityAttribute, List<AttributeInstanceResult>>` as it is your common structure between the two buildXXX functions

Comment (HoucemKacem) [reply search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityService.kt]: Done

File: search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityService.kt (hunk 2 above)
Comment (HoucemKacem) [reply search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityService.kt]: Done

**Review by bobeal — APPROVED** [review]

**Pull request merged**